File: examples/testclient.js
"use strict";

let http = require('http');
let url = require('url');
let request = require('request');
let concatStream = require('concat-stream');

const SEND_URL = 'http://localhost:9004/send';

function sendMessage(message, to) {
    request({
        url: SEND_URL,
        method: 'POST',
        json: true,
        body: {
            message: message,
            to: to
        },
        headers: {
            'Content-Type': 'application/json'
        }
    }, (err, response, body) => {
        if (err) {
            console.log('Failed to send messsage!');
            console.log(err);
        } else {
            console.log(
                'Response to send request:\n\tSuccess:', body.success, 
                '\n\tQueue Index:', body.queueIndex);
        }
    });
}

function handleBody(req, callback) {
  req.on('error', function (err) {
    callback(err, null);
  });
  var concat = concatStream(function (data) {
    callback(null, data);
  });
  req.pipe(concat);
}

let server = http.createServer((req, res) => {
    if (url.parse(req.url).path === '/received') {
        handleBody(req, (err, body) => {
            if (err) {
                console.log('Couldn\'t read POST body\n', err);
            } else {
                console.log('Got a message!');
                console.log(body.toString());
            }
            res.write('Thanks!');   
            res.end();
            server.close();
        });
    } else {
        console.log('Got an unknown request for', req.url);
        res.write('What?');
        res.end();
    }
});

server.listen(9005);
console.log('Listening on port 9005');

// Send a message to chamber to get the test started

sendMessage('Hello, world!', 'myself');

File: examples/testclient.py
import json
from flask import Flask, request
import requests as req

app = Flask(__name__)

SEND_URL = 'http://localhost:9004/send'

def send_message(message, to):
    json_data = {'message': message, 'to': to}
    response = req.post(SEND_URL, json=json_data)
    data = json.loads(response.text)
    print('Response to send request:\n\tSuccess: {0}\n\tQueue Index: {1}'.format(
        data['success'], data['queueIndex']))

@app.route('/received', methods=['POST'])
def received():
    print('Got message!')
    print(request.params)
    return 'Thanks!'

@app.route('/')
def unknown():
    return 'What?'

# Send a message to chamber to get the test started
send_message('Hello, world!', 'myself')
app.run(port=9005)

